source/dub/packagesupplier.d

```diff
@@ -146,6 +146,50 @@ class FileSystemPackageSupplier : PackageSupplier {
 	}
 }
 
+/**
+	Git based based package supplier.
+
+	This package supplier downloads directly from git.
+*/
+class GitPackageSupplier : PackageSupplier {
+	this() { }
+
+	override @property string description() { return typeof(this).stringof; }
+
+	Version[] getVersions(string package_id)
+	{
+		return null;
+	}
+
+	void fetchPackage(NativePath path, string packageId, Dependency dep, bool pre_release)
+	{
+		enforce(path.absolute);
+		logInfo("Storing package '"~packageId~"', version requirements: %s", dep);
+		auto gitDep=dep.gitDep;
+		assert(gitDep.valid);
+		URL url = gitDep.toURLZip;
+		logDiagnostic("Downloading from '%s'", url);
+		// TODO: foreach(i; 0..3)
+		download(url, path);
```

Comment:
Might be worthwhile to generalize the retry logic in `download` (almost all supplier seem to require this)